library/components/TUrlCopyButton.vue

```diff
@@ -1,57 +1,75 @@
 <template>
-    <div class="relative">
-        
-        <t-tooltip position="bottom" width="130px">
-            <button
-                class="h-[30px] w-[30px] rounded-[4px] flex items-center justify-center border border-[#C3C2CB] p-1 transition-colors"
-                @click="copy"
-                :class="{ 'bg-[#B4E4D9]': copied }"
-                slot="trigger"
-            >
-                <check-icon :size="18" v-if="copied" />
-                <link-icon :size="18" v-else />
-                <span class="visually-hidden">{{ copied ? "URL Copied" : "Copy URL" }}</span>
-            </button>
-            {{ copied ? "URL Copied" : "Copy URL" }}
-        </t-tooltip>
-    </div>
+  <div class="relative">
+    
+    <t-tooltip position="bottom" width="130px">
+      <button
+        class="h-[30px] w-[30px] rounded-[4px] flex items-center justify-center border border-[#C3C2CB] p-1 transition-colors"
+        @click="copy"
+        :class="{ 'bg-[#B4E4D9]': copied }"
+        slot="trigger"
+      >
+        <check-icon :size="18" v-if="copied" />
+        <link-icon :size="18" v-else />
+        <span class="visually-hidden">{{ copied ? "URL Copied" : "Copy URL" }}</span>
+      </button>
+      {{ copied ? "URL Copied" : "Copy URL" }}
+    </t-tooltip>
+  </div>
 </template>
 
 <script>
-    export default {
-        data: () => ({
-            copied: false,
-        }),
-        methods: {
-            copy() {
-                this.copied = true;
-                const tempInput = document.createElement('input'),
-                text = window.location.href;
-                document.body.appendChild(tempInput);
-                tempInput.value = text;
-                tempInput.select();
-                document.execCommand('copy');
-                document.body.removeChild(tempInput);
-                setTimeout(() => {
-                    this.copied = false;
-                }, 1000)
-            }
+  export default {
+    data: () => ({
+      copied: false,
+    }),
+    computed: {
+      url() {
+        const parentUrl = this.filters['context[parent]'];
+
+        if (parentUrl) {
+          const pageContext = 'context[page]=' + this.page.url;
+          return parentUrl + "?" + pageContext + "&" + this.urlParams;
         }
+
+        return window.location.href;
+      },
+      urlParams() {
+        return new URLSearchParams(this.filters).toString();
+      }
+    },
+    methods: {
+      copy() {
+        this.copied = true;
+
+        const tempInput = document.createElement('input');
+        document.body.appendChild(tempInput);
+        tempInput.value = this.url;
+        tempInput.select();
+
+        document.execCommand('copy');
```

Comment:
This is kind of old API to create this kind of interactions. We should be using the new clipboard API instead: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard

Here we have an example: https://googlechrome.github.io/samples/async-clipboard/

And the API is Promise based so we can set the `copied` attribute in the `then` instead of using the `setTimeout` that you added on line 52

Reply:
@egcastro Clipboard API is blocked under iframe, due to security permission policy.

Reply:
Ok you're right. So the way we should be doing this is by creating a new message and `postMessage` a command that says `copy` with the string we want to be copied and we should use the clipboard API in the parent iframe.

If we are not willing to do this because of any reason, we should check browser compatibility because the execCommand API is deprecated and it has poor cross-browser support.

<img width="815" alt="image" src="https://user-images.githubusercontent.com/1915140/189614638-15c2c194-5a44-4a11-9bf7-039f4035da6c.png">

Reply:
I see, let's go for this approach for now, then we can create the PR in registry to handle such operation.
Then it should be like this.
1. Copy base url if not in app.snyk.io
2. Send message to parent if there is a context.

This might reduce the url size because then we do not need a full url as context, instead it can be just `context[parent]=true`

Reply:
Well, but to keep it consistent we should have a method that hides the logic for components. We can create the method `copyToClipboard` for instance in the visualization_mixin.
There we should use the new `Clipboard` API if we are not inside an iframe or use the `postMessage` otherwise.
This way we hide all the logic of `iframe` vs `not-iframe` out of components. So that it's easier and centralised when we have a version that can run outside of iframes.

Reply:
In this particular case, the component is library component, and sure other components might use copy logic as well, so probably makes sense to take it out of this component and copy to `common_mixin`, I'll see if it works good that way and split the PR.

Because Ezra probably needs this merged today, we can delay postMessage change for now.

Reply:
Got it, in that case let's leave everything in this component for now 😅, we will move to core with final version of this function. Thanks.